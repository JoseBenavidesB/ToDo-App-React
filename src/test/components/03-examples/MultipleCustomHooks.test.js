import React from 'react'
import { shallow } from 'enzyme';
import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks';
import { useFetch } from '../../../hooks/useFetch';
import { useCounter } from '../../../hooks/useCounter';
jest.mock('../../../hooks/useFetch'); // SIMULACION con esto evito que se ejecute el customhooks y en cambio se ejecuta lo que yo voy hacer
jest.mock('../../../hooks/useCounter');


describe( 'test MultipleCustomHooks ', () => {

    useCounter.mockReturnValue({
        counter: 10,
        increment: () => {},
        decrement: () => {}
    });

    test('should return correctly', () => { 

        useFetch.mockReturnValue({ //simulacion useFetch con MOCK
            data: null,
            loading: true,
            error: null
        });

        const wrapper = shallow( <MultipleCustomHooks /> );
        expect(wrapper).toMatchSnapshot();

     });

     test('should show data', () => {

        useFetch.mockReturnValue({
            data: [{
                author: 'jose',
                quote: 'hello world',
            }],
            loading: false,
            error: null
        });
        
        const wrapper = shallow( <MultipleCustomHooks /> );

        expect( wrapper.find('.alert').exists() ).toBe( false );
        expect( wrapper.find('.mb-3').text().trim() ).toBe( 'hello world');
        expect( wrapper.find('footer').text().trim() ).toBe( 'jose');

     });
})