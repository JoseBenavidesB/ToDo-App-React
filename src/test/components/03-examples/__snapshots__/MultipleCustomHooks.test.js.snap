// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test MultipleCustomHooks  should return correctly 1`] = `
<Fragment>
  <h1>
     BreakingBad 
  </h1>
   
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loanding....
  </div>
  <button
    className="btn btn-success"
    onClick={[Function]}
  >
     Previous Quote 
  </button>
  <button
    className="btn btn-success"
    onClick={[Function]}
  >
     Next Quote 
  </button>
</Fragment>
`;
